### ISSUE TITLE
Incompatible with newest VTest

### ISSUE THREAD

**Issue by TimWolla** [haproxy/haproxy#1349]
### Tool Name and Version

VTest

### Code Report

```plain
**   top   === client c_err2 -connect ${h1_err2h1_sock} {
  **   c_err2 Starting client
  **   c_err2 Waiting for client
  **   c_err2 Started on 127.0.0.1:39139 (1 iterations)
  ***  c_err2 Connect to 127.0.0.1:39139
  ***  h1    debug|0000000a:err2.accept(000b)=0013 from [127.0.0.1:36630] ALPN=<none>
  ***  c_err2 connected fd 13 from 127.0.0.1 36630 to 127.0.0.1:39139
  **   c_err2 === send "PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n"
  **** c_err2 send|PRI * HTTP/2.0\r
  **** c_err2 send|\r
  **** c_err2 send|SM\r
  **** c_err2 send|\r
  **   c_err2 === expect_close
  **** c_err2 Expecting close (fd = 13)
  **** dT    0.011
  ***  h1    debug|0000000a:err2.clireq[0013:ffffffff]: PRI * HTTP/2.0
  ---- c_err2 Expecting close: read = 1, c = 0x48
  ***  h1    debug|0000000a:err2.clicls[0013:ffffffff]
  ***  h1    debug|0000000a:err2.closed[0013:ffffffff]
  *    top   RESETTING after reg-tests/connection/tcp_to_http_upgrade.vtc
  **   h1    Reset and free h1 haproxy 9690
  **   h1    Wait
  **   h1    Stop HAproxy pid=9690
  **** h1    Kill(2)=0: Success
  **** h1    STDOUT EOF
  **** dT    0.111
  **   h1    WAIT4 pid=9690 status=0x0002 (user 0.002960 sys 0.002960)
  **   s1    Waiting for server (4/-1)
  *    top   TEST reg-tests/connection/tcp_to_http_upgrade.vtc FAILED
```

### Additional Information

VTest was updated today, apparently it broke some of the tests: https://github.com/vtest/VTest/commits/master

Failed CI: https://github.com/haproxy/haproxy/actions/runs/1136153978

### Output of `haproxy -vv`

```plain
HAProxy version 2.5-dev3 2021/08/01 - https://haproxy.org/
Status: development branch - not safe for use in production.
Known bugs: https://github.com/haproxy/haproxy/issues?q=is:issue+is:open
Running on: Linux 5.8.0-1039-azure #42~20.04.1-Ubuntu SMP Thu Jul 15 14:11:07 UTC 2021 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference -Werror
  OPTIONS = 
  DEBUG   = -DDEBUG_STRICT=1

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT -PCRE2 -PCRE2_JIT +POLL +THREAD +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO -OPENSSL -LUA +ACCEPT4 -CLOSEFROM -ZLIB +SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL -SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS -OT -QUIC -PROMEX -MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=2).
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built without PCRE or PCRE2 support (using libc's regex instead)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 9.3.0

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : none

Available filters :
	[SPOE] spoe
	[CACHE] cache
	[FCGI] fcgi-app
	[COMP] compression
	[TRACE] trace
```

**Comment by chipitsine** [comment]
as we use consistent way to install VTest

https://github.com/haproxy/haproxy/blob/master/scripts/build-vtest.sh

we may add "if VTest is pinned, install it" and pin to commit hash using github variables

**Comment by chipitsine** [comment]
if nobody minds, I'll modify build-vtest.sh accordingly

**Comment by TimWolla** [comment]
I mind.

Modifying GitHub Variables is intransparent and requires Willy to click around in the web interface. Sending a patch to pin + a revert to unpin is much easier.

Also it's not even clear whether this is a bug in VTest or a bug in our vtc that was exposed by some VTest change.

**Comment by TimWolla** [comment]
Fixed with ab4fa24cd832bfd22b06c0dd49db8dfee3b69cc4.
